### PR TITLE
Make openssl/rsa.h coexist with complex.h.

### PR DESCRIPTION
**Pull request by paulidale**
Use 'i' as parameter name not 'I'.
The latter causes problems when complex.h is #included.

- [x] documentation is added or updated
- [x] tests are added or updated

Fixes: #7224

### CODE REVIEW

**Comment by levitte** [comment]
Wait hold on, Travis fault relevant

**Comment by paulidale** [comment]
Fix for Travis is in, waiting on the builds.

File: test/rsa_complex.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/*
+ * Check to see if there is a conflict between complex.h and openssl/rsa.h.
+ * the former defines "I" as a macro and earlier versions of the latter use
+ * for funtion arguments.
```
Comment (romen) [inline test/rsa_complex.c]: nit: typo

File: test/rsa_complex.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# if __STDC_VERSION__ >= 199901L
+#  include <complex.h>
+# endif
+#endif
+#include <openssl/rsa.h>
+#include <stdlib.h>
+
+int main(int argc, char *argv[])
+{
+    /* There are explictly no run time checks for this one */
```
Comment (romen) [inline test/rsa_complex.c]: typo: explicitly

File: test/rsa_complex.c (hunk 1 above)
Comment (paulidale) [reply test/rsa_complex.c]: Thanks, fixed this and the capitalisation problem.

**Comment by paulidale** [comment]
Merged, thanks for the corrections.

**Comment by paulidale** [comment]
Should this go to 1.1.1?

**Comment by levitte** [comment]
+1 for cheery picking to both 1.1.1 and 1.1.0

**Comment by paulidale** [comment]
Merged to 1.1.0 and 1.1.1.

File: test/rsa_complex.c [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+/*
+ * Check to see if there is a conflict between complex.h and openssl/rsa.h.
+ * The former defines "I" as a macro and earlier versions of the latter use
+ * for function arguments.
+ */
+#if defined(__STDC_VERSION__)
+# if __STDC_VERSION__ >= 199901L
+#  include <complex.h>
```
Comment (gvanem) [inline test/rsa_complex.c]: For DJGPP, a `<complex.h>` is available only for C++. But this creates a `rsa_complex.exe`:
```diff
--- a/test/rsa_complex.c  2018-12-21 22:13:43
+++ b/test/rsa_complex.c 2018-12-13 16:44:01
@@ -13,7 +13,7 @@
  * for function arguments.
  */
 #if defined(__STDC_VERSION__)
-# if (__STDC_VERSION__ >= 199901L) && !defined(__DJGPP__)
+# if __STDC_VERSION__ >= 199901L
 #  include <complex.h>
 # endif
 #endif
```

(too lazy to create yet another PR for this).

**Pull request closed without merging**